src/netex_idf/lines.rs

```diff
@@ -188,8 +214,15 @@ fn make_lines(lines_netex_idf: &CollectionWithId<LineNetexIDF>) -> Result<Collec
         let codes: KeysValues = ln
             .private_code
             .clone()
-            .map(|pc| vec![("Netex_PrivateCode".into(), pc)].into_iter().collect())
-            .unwrap_or_else(BTreeSet::new);
+            .map(|pc| {
+                vec![
+                    ("source".into(), ln.source_code.clone()),
+                    ("Netex_PrivateCode".into(), pc),
+                ]
+                .into_iter()
+                .collect()
+            })
+            .unwrap_or_else(KeysValues::default);
```

Comment:
You cannot make the insertion of source_code (necessarily existing) dependent by private_code (optional)

Reply:
Nice catch indeed. Something like the following should be more correct I believe.

```rust
let mut codes = KeysValues::default();                       
codes.insert(("source".into(), ln.source_code.clone()));     
if let Some(private_code) = ln.private_code.clone() {        
    codes.insert(("Netex_PrivateCode".into(), private_code));
}                                                            
```

Reply:
Done in 023fca7b89a26e414ed850d20752eba0fc0513eb.